### PR TITLE
Hb 04

### PR DESCRIPTION
**Pull request by Alex-Korzun**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> getAllSessionsQuery =
+                    session.createQuery("SELECT ms FROM MovieSession ms "
+                                    + "LEFT JOIN FETCH ms.cinemaHall LEFT JOIN FETCH ms.movie "
+                                    + "WHERE ms.movie.id = :id_movie "
+                                    + "AND DATE_FORMAT(ms.showTime, '%Y-%m-%d') = :date ",
+                            MovieSession.class);
+            getAllSessionsQuery.setParameter("id_movie", movieId);
+            getAllSessionsQuery.setParameter("date",
+                    DateTimeFormatter.ISO_LOCAL_DATE.format(date));
```
Comment (Likh-Alex) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java:51]: Just as an opinion - I prefer to chain the methods and avoid redundant variables)

**Review by Likh-Alex — APPROVED** [review]

File: src/main/java/com/dev/cinema/model/MovieSession.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "movieSession")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/model/MovieSession.java]:
```suggestion
@Table(name = "movie_session")
```

File: src/main/java/com/dev/cinema/model/CinemaHall.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
+package com.dev.cinema.model;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "cinemaHall")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/model/CinemaHall.java]:
```suggestion
@Table(name = "cinema_hall")
```

File: src/main/java/com/dev/cinema/model/MovieSession.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    @JoinColumn(name = "movie_id")
+    private Movie movie;
+    @ManyToOne
+    @JoinColumn(name = "cinemaHall_id")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/model/MovieSession.java]:
```suggestion
    @JoinColumn(name = "cinema_hall_id")
```

**Review by dmytro-hryhoruk — APPROVED** [review]
Good job!

File: src/main/java/com/dev/cinema/model/MovieSession.java (hunk 2 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/model/MovieSession.java]: done

File: src/main/java/com/dev/cinema/model/CinemaHall.java (hunk 3 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/model/CinemaHall.java]: done

File: src/main/java/com/dev/cinema/model/MovieSession.java (hunk 4 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/model/MovieSession.java]: done

**Pull request merged**